tests/integration/security/file_mounted_certs/main_test.go

```diff
@@ -0,0 +1,206 @@
+      "ISTIO_META_TLS_CLIENT_CERT_CHAIN": "/client-certs/cert.pem",
+      "ISTIO_META_TLS_CLIENT_KEY": "/client-certs/key.pem",
+      "ISTIO_META_TLS_CLIENT_ROOT_CERT": "/client-certs/ca.pem",
+      "ISTIO_META_TLS_SERVER_CERT_CHAIN": "/server-certs/cert.pem",
+      "ISTIO_META_TLS_SERVER_KEY": "/server-certs/key.pem",
+      "ISTIO_META_TLS_SERVER_ROOT_CERT": "/server-certs/ca.pem",
+      "ISTIO_META_PROXY_XDS_VIA_AGENT": "false",
+      "ISTIO_METAJSON_METRICS_INCLUSIONS": '{\"sidecar.istio.io/statsInclusionPrefixes\": \"access_log_file,cluster,cluster_manager,control_plane,http,http2,http_mixer_filter,listener,listener_manager,redis,runtime,server,stats,tcp,tcp_mixer_filter,tracing\"}',
+      "ISTIO_META_DNS_CAPTURE": "false",
+	}
+`
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+
+		// SDS requires Kubernetes 1.13
+		RequireEnvironmentVersion("1.13").
+		Label("CustomSetup").
+		Setup(istio.Setup(&inst, setupConfig, CreateCustomIstiodSecret)).
+		Run()
+}
+
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+
+	cfg.ControlPlaneValues = `
+components:
+  egressGateways:
+  - enabled: false
+    name: istio-egressgateway
+  ingressGateways:
+  - enabled: false
+    name: istio-ingressgateway
+  pilot:
+    enabled: true
+    k8s:
+      overlays:
+        - kind: Deployment
+          name: istiod
+          patches:
+            - path: spec.template.spec.containers.[name:discovery].args
+              value: [
+                "discovery",
+                "--monitoringAddr=:15014",
+                "--log_output_level=default:info",
+                "--domain",
+                "cluster.local",
+                "--trust-domain=cluster.local",
+                "--keepaliveMaxServerConnectionAge",
+                "30m",
+                "--resync",
+                "5m",
+                "--caCertFile",
+                "/server-certs/ca.pem",
+                "--tlsCertFile",
+                "/server-certs/cert.pem",
+                "--tlsKeyFile",
+                "/server-certs/key.pem"
+              ]
+            - path: spec.template.spec.volumes[8]
+              value: |-
+                name: server-certs
+                secret:
+                  secretName: test-istiod-server-cred
+                  defaultMode: 420
+            - path: spec.template.spec.containers[name:discovery].volumeMounts[7]
+              value: |-
+                name: server-certs
+                mountPath: /server-certs
+
+meshConfig:
+  defaultConfig:
+    controlPlaneAuthPolicy: "NONE"
+    discoveryAddress: istiod.istio-system:15010
+    proxyMetadata: ` + strings.Replace(ProxyMetadataJson, "\n", "", -1) +
+`
+values:
+  global:
+    pilotCertProvider: "mycopki"
+    logging:
+      level: "default:debug"
+    proxy:
+      logLevel: debug
+      componentLogLevel: "misc:debug"
+  pilot:
+    env:
+      PILOT_SKIP_VALIDATE_TRUST_DOMAIN: "true"
```

Comment:
why are we skipping?

Reply:
The test certificates that are used as server/client certs don't have `spiffe://cluster.local` in SANs; and if Trust Domain validation is not skipped, test Echo server will refuse connections due to:
```
            "combined_validation_context": {
             "default_validation_context": {
              "match_subject_alt_names": [
               {
                "prefix": "spiffe://cluster.local/"
               }
              ]
             },

```

I looked into this a bit, and It seems that i can try using the `default` certs as "client certs". Those have the Spiffe URI in SANs. Let me make that change.

Reply:
fixed